pangeo_forge_recipes/storage.py

```diff
@@ -116,14 +115,15 @@ def size(self, path: str) -> int:
         return self.fs.size(self._full_path(path))
 
     @contextmanager
-    def open(self, path: str, **kwargs) -> Iterator[None]:
+    def open(self, path: str, **kwargs) -> Iterator[fsspec.core.OpenFile]:
         """Open file with a context manager."""
         full_path = self._full_path(path)
         logger.debug(f"entering fs.open context manager for {full_path}")
-        with self.fs.open(full_path, **kwargs) as f:
-            logger.debug(f"FSSpecTarget.open yielding {f}")
-            yield f
-            logger.debug("FSSpecTarget.open yielded")
+        of = self.fs.open(full_path, **kwargs)
+        logger.debug(f"FSSpecTarget.open yielding {of}")
+        yield of
+        logger.debug("FSSpecTarget.open yielded")
+        of.close()
```

Comment:
Correction: self.fs.open returns a file-like object (io.IOBase subclass, AbstractBufferedFile for s3, gcs, etc.)

For an AbstractBufferedFile, entering the context is a no-op yielding the same object, and exiting calls `.close()`. For other file types, it's something else, with the expectation that exiting the context closes. For example, LocalFileSystem.open -> LocalFileOpener ; LocalFileSystem.__enter__ -> _io.BufferedReader (and exit calls the actual file's exit, calling close()). You *could* do `with` on the file-like.

For fsspec.open, you get OpenFile objects, which can contain more than one file-like layer, for compression and text mode. You must use this with a context, to get the outermost file-like object back.  Leaving the context calls close on the file-like objects in reverse order. If you, instead, call `.open` of the OpenFile, you get the same file-like object, but its .close method has been patched to execute the same chain of close calls, since this object might in this case outlive the OpenFile instance that made it.